include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -417,6 +417,53 @@ def MulFOpIEEE754 : ArithBinaryFloatingPointLibraryOp<"ieee754.mul", [
   }];
 }
 
+// This models the division and square root, distinguished by `sqrtOp`,
+// operation interface in Berkeley HardFloat. It computes `left`/`right`
+// if `sqrtOp` = 0; it ignores `right` and computes sqrt(`left`) if `sqrtOp` = 1.
+def DivSqrtOpIEEE754 : ArithBinaryFloatingPointLibraryOp<"ieee754.divSqrt", [
+    SameTypeConstraint<"left", "out">
```

Comment:
Should this also apply to `left` and `right`?

Reply:
Yes, and it's already asserted in its parent hierarchy: https://github.com/llvm/circt/blob/dd4b9c0dba6a1ed62e724646c2381b0e913aca42/include/circt/Dialect/Calyx/CalyxPrimitives.td#L338

Reply:
Sweet thanks!